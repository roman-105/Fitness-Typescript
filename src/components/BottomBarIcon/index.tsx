import React from 'react';
import { HomeIcon, ExploreIcon, ProgressIcon, CoachIcon, MyClubIcon } from '../Icon';
import Typography from '../Typography/Typography';
import styles from '../Typography/typography-styles';
import formatMessage from 'format-message';
import { Routes } from '../../router/routes';

const lookup: { [index: string]: (color: string) => JSX.Element } = {
  [Routes.Home]: (color: string) => <HomeIcon fill={color} />,
  [Routes.Programs]: (color: string) => <ExploreIcon fill={color} />,
  [Routes.Progress]: (color: string) => <ProgressIcon fill={color} />,
  [Routes.Coach]: (color: string) => <CoachIcon fill={color} />,
  [Routes.Club]: (color: string) => <MyClubIcon fill={color} />
};

export function bottomBarIcon(name: Routes) {
  return function ({ color }: { color: string }): JSX.Element {
    return lookup[name](color);
  };
}

export function bottomBarLabel(name: Routes) {
  const tabNameLookup: { [index: string]: string } = {
    [Routes.Home]: formatMessage('Home'),
    [Routes.Programs]: formatMessage('Explore'),
    [Routes.Progress]: formatMessage('Progress'),
    [Routes.Coach]: formatMessage('Coach'),
    [Routes.Club]: formatMessage('Clubs')
  };

  return function ({ color }: { color: string }): JSX.Element {
    return (
      <Typography
        style={{ color, ...styles.label }}
        fontFamily="regular"
        fontSize={11}
        lineHeight={12}
      >
        {tabNameLookup[name]}
      </Typography>
    );
  };
}
