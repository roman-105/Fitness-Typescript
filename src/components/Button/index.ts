export { default as BFButton } from './BFButton';
export { default as BFSecondaryButton } from './BFSecondaryButton';
export { default as BFSelectorButton } from './selector/BFSelectorButton';
export { default as BFLightButton } from './BFLightButton';
