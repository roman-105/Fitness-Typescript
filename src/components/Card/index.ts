export { default as MediaCard } from './Media/MediaCard';
export { default as SlashedCard } from './SlashedCard/SlashedCard';
