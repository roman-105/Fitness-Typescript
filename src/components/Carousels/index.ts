export { default as MediaCardCarousel } from './MediaCardCarousel';
import { MediaCardCarouselProps as mediaprops } from './MediaCardCarousel';
export { default as ImageCarousel } from './ImageCarousel';
import { ImageCarouselProps as imageprops } from './ImageCarousel';
export type MediaCardCarouselProps = mediaprops;
export type ImageCarouselProps = imageprops;
