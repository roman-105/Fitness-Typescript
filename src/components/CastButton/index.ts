export { default as BFCastButton } from './BFCastButton';
