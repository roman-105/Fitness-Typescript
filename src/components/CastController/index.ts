export { default as CastController } from './CastController';
