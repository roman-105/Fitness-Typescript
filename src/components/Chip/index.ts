export { default as ChipList } from './ChipList';
export { default as ChipItem } from './ChipItem';
import { Item } from './ChipList';
export type ChipItemType = Item;
