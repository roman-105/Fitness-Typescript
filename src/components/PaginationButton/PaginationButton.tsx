import React from 'react';
import { View } from 'react-native';
import Ripple from 'react-native-material-ripple';
import Typography from '../Typography';
import styles from './pagination-button-styles';
import formatMessage from 'format-message';

export interface PaginationButtonProps {
  content: string;
  onPress: () => void;
  hasMore: boolean;
  hide?: boolean;
}

function PaginationButton({
  content,
  onPress,
  hide,
  hasMore
}: PaginationButtonProps): JSX.Element | null {
  if (hide) return null;

  const handleOnPress = () => {
    setTimeout(function () {
      onPress();
    }, 200);
  };

  if (hasMore) {
    return (
      <Ripple onPress={handleOnPress} style={styles.button} testID="pagination-button-ripple">
        <Typography fontFamily="regular" fontSize={14}>
          {formatMessage('Show more')}
        </Typography>
      </Ripple>
    );
  } else {
    return (
      <View style={styles.container}>
        <Typography fontFamily="regular" fontSize={14}>
          {formatMessage('There are no more {content} to show.', { content })}
        </Typography>
      </View>
    );
  }
}

export default PaginationButton;
