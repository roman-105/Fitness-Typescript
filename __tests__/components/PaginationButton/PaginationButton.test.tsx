import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { PaginationButton } from '../../../src/components/PaginationButton';

describe('PaginationButton', () => {
  it.skip('when it can show more renders correctly and click on it triggers onPress func', async () => {
    // Ripple not captured
    const onPressFn = jest.fn();
    const { getByText } = render(
      <PaginationButton hasMore={true} content="recipe" onPress={onPressFn} />
    );

    const textElement = getByText('Show more');
    fireEvent.press(textElement);

    expect(textElement).not.toBeNull();
    expect(onPressFn).toHaveBeenCalledTimes(1);
  });

  it('when it can not show more renders correctly and click on it not triggers onPress func', async () => {
    const onPressFn = jest.fn();
    const { getByText } = render(
      <PaginationButton hasMore={false} content="recipe" onPress={onPressFn} />
    );

    const textElement = getByText('There are no more recipe to show.');
    expect(textElement).not.toBeNull();
    expect(onPressFn).toHaveBeenCalledTimes(0);
  });
  it('dosent show when we hide it', async () => {
    const onPressFn = jest.fn();
    const { toJSON } = render(
      <PaginationButton hide hasMore={false} content="recipe" onPress={onPressFn} />
    );
    expect(toJSON()).toBeNull();
  });
  it('passes snapshot testing on hasmore true', async () => {
    const onPressFn = jest.fn();
    const { toJSON } = render(
      <PaginationButton hasMore={true} content="recipe" onPress={onPressFn} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('passes snapshot testing on hasmore false', async () => {
    const onPressFn = jest.fn();
    const { toJSON } = render(
      <PaginationButton hasMore={false} content="recipe" onPress={onPressFn} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
