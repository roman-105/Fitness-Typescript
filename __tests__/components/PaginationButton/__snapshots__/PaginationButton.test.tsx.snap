// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton passes snapshot testing on hasmore false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Argumentum",
        },
        undefined,
        Object {
          "fontSize": 14,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    There are no more recipe to show.
  </Text>
</View>
`;

exports[`PaginationButton passes snapshot testing on hasmore true 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "height": 48,
      "justifyContent": "center",
    }
  }
  testID="pagination-button-ripple"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Argumentum",
        },
        undefined,
        Object {
          "fontSize": 14,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Show more
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  />
</View>
`;
